lensit/__init__.py

```diff
@@ -295,15 +295,25 @@ def get_maps_lib(exp, LDres, HDres=14, cache_lenalms=True, cache_maps=False, \
             pix_pha=pixpha, cache_sims=cache_maps)
 
 
-def get_isocov(exp, LD_res, HD_res=14, pyFFTWthreads=4, fn_tensCls=None):
+def get_isocov(exp, LD_res, HD_res=14, pyFFTWthreads=4, do_tensor = False, \
+    fn_scalCls=None, fn_lensCls=None, fn_tensCls=None, r=0.0, lmin=0): 
     """
     Set HD_res to 14 for full sky sampled at res LD.
     """
-    sN_uKamin, sN_uKaminP, Beam_FWHM_amin, ellmin, ellmax = get_config(exp)
-    if fn_tensCls is None:
-        cls_unl, cls_len = get_fidcls(ellmax_sky=ellmax_sky)
+    sN_uKamin, sN_uKaminP, Beam_FWHM_amin, ellmin, ellmax = get_config(exp) 
+    ellmin = lmin 
+
+    if do_tensor:
+        scalCls_unl = get_tenscls(fn_scalCls, ellmax_sky=ellmax_sky)
+        scalCls_len = get_tenscls(fn_lensCls, ellmax_sky=ellmax_sky)
+        tensCls = get_tenscls(fn_tensCls, ellmax_sky=ellmax_sky)
```

Comment:
It sounds like you should just change the name of get_tenscls to get_cls_from_fn or something like that, since you are using it to get scalCls and lensCls as well. Also inside get_tenscls, you have a flag for type='tenscls' is that still gonna work with scalCls and lensCls? (Probably is just saying it's 4 columns of TT, TE, EE, BB, may be worth double checking.)